### ISSUE TITLE
CECO compsets have runtime error with gnu compiler

### ISSUE THREAD

**Issue by mnlevy1981** [marbl-ecosys/MARBL#40]
When building CESM with the gnu compiler, compsets that use the ecosystem ocean tracer module produce the following runtime error:

```
     (passive_tracer_tools:read_field_3D) reading alk_riv_flux from /glade/p/cesmdata/cseg/inputdata/ocn/pop/gx1v6/forcing/river_nutrients_GNEWS2000_gx1v6.nc
     (passive_tracer_tools:read_field_3D) reading doc_riv_flux from /glade/p/cesmdata/cseg/inputdata/ocn/pop/gx1v6/forcing/river_nutrients_GNEWS2000_gx1v6.nc
 (tavg_requested) id =            0
     (tavg_requested) FATAL ERROR: invalid tavg id
------------------------------------------------------------------------

POP aborting...
 FATAL ERROR: invalid tavg id

------------------------------------------------------------------------
```

The error does not crop up when using the Intel or NAG compilers.

**Comment by klindsay28** [comment]
This appears to be a problem in one of the source files comprising the POP driver for MARBL, not in MARBL itself. The patch below, to POP, enables CECO and GECO compsets to run with gnu. However, even with this patch, tests with ciso enabled are failing with gnu.

# Index: ecosys_tavg.F90

--- ecosys_tavg.F90 (revision 78054)
+++ ecosys_tavg.F90 (working copy)
@@ -235,7 +235,7 @@

```
 nblocks_clinic = size(flux_diags,4)
```
-    !$OMP PARALLEL DO PRIVATE(iblock)
-    !$OMP PARALLEL DO PRIVATE(iblock,i)
   do iblock=1,nblocks_clinic
  
  ```
  associate (diag_cnt => marbl_instances(iblock)%surface_forcing_diags%diag_cnt)
  ```

**Comment by klindsay28** [comment]
Whoops, I didn't preview, so I didn't see what github was going to do to cut-n-paste of the patch. My bad. I've attached a .txt file of the patch this time. The fix is included as a patch because the code in question is outside of git.
[patch.txt](https://github.com/NCAR/MARBL/files/233275/patch.txt)

**Comment by mnlevy1981** [comment]
This fix is now in both the POP trunk and the `marbl_dev` branch of POP; additionally, an issue that prevented running CISO with GNU or NAG has been fixed via pull request #54
